src/test/java/org/jenkinsci/plugins/docker/workflow/WithContainerStepTest.java

```diff
@@ -236,4 +236,38 @@
         });
     }
 
+    @Test public void defaultEntrypoint() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                DockerTestUtil.assumeDocker();
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "prj");
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                        "  withDockerContainer('ubuntu') {\n" +
+                        "    sh 'cat /proc/1/cmdline'\n" +
+                        "  }\n" +
+                        "}", true));
+                WorkflowRun b = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+                story.j.assertLogContains("cat", b);
+            }
+        });
+    }
+
+    @Test public void allowRunOverrides() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                DockerTestUtil.assumeDocker();
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "prj");
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                        "  withDockerContainer(args: '--entrypoint top', image: 'ubuntu') {\n" +
```

Comment:
I also have this use case. We have images that use `ENTRYPOINT` to launch background processes via supervisord or runit. These processes are used by the commands within the `inside` block. Since the changes in #63 we've had to stay pinned to 0.7.